### PR TITLE
revert call shutdown in LifecycleNode destructor

### PR DESCRIPTION
**Pull request by fujitatomoya**
revert https://github.com/ros2/rclcpp/pull/2528 and https://github.com/ros2/rclcpp/pull/2545

backports are not straight-forward because of the conflicts, so end up having backport PRs for each distro.

- Jazzy: https://github.com/ros2/rclcpp/pull/2558
- Iron: https://github.com/ros2/rclcpp/pull/2559
- Humble: https://github.com/ros2/rclcpp/pull/2560

### CODE REVIEW

File: rclcpp_lifecycle/src/lifecycle_node.cpp [hunk 1]
```diff
@@ -152,37 +152,8 @@ LifecycleNode::LifecycleNode(
...
-  } else if (current_state > lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED) {
-    RCLCPP_WARN(
-      rclcpp::get_logger("rclcpp_lifecycle"),
-      "Shutdown error in destruction of LifecycleNode: Node still in transition state(%u)",
-      current_state);
-  }
-
   // release sub-interfaces in an order that allows them to consult with node_base during tear-down
   node_waitables_.reset();
-  node_type_descriptions_.reset();
```
Comment (fujitatomoya) [inline rclcpp_lifecycle/src/lifecycle_node.cpp:185]: this is actually bug fix, i will re-submit this cosmetic after revert as follow-up.

File: rclcpp_lifecycle/src/lifecycle_node.cpp [hunk 2]
```diff
@@ -191,7 +162,6 @@ LifecycleNode::~LifecycleNode()
   node_timers_.reset();
   node_logging_.reset();
   node_graph_.reset();
-  node_base_.reset();
```
Comment (fujitatomoya) [inline rclcpp_lifecycle/src/lifecycle_node.cpp:194]: this is actually bug fix, i will re-submit this cosmetic after revert as follow-up.

File: rclcpp_lifecycle/test/test_lifecycle_node.cpp [hunk 3]
```diff
@@ -447,146 +447,6 @@ TEST_F(TestDefaultStateMachine, bad_mood) {
   EXPECT_EQ(1u, test_node->number_of_callbacks);
 }
 
-
-TEST_F(TestDefaultStateMachine, shutdown_from_each_primary_state) {
```
Comment (fujitatomoya) [inline rclcpp_lifecycle/test/test_lifecycle_node.cpp:451]: this is test enhancement, i will re-submit this cosmetic after revert as follow-up.

**Comment by fujitatomoya** [comment]
CI (`--packages-above-and-dependencies rclcpp_lifecycle`):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21242/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15562/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/721/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21964/)

**Review by wjwwood — APPROVED** [review]

**Pull request merged**